primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -132,6 +129,16 @@ public static boolean isSystemWindows() {
         return File.separatorChar == '\\';
     }
 
+    public static int containsUnprintableCharacters(String s) {
```

Comment:
This replaces what used to do SafeFile?

Reply:
Yes all your other checks look like they already cover the other checks from what SafeFile was doing.

Reply:
There is one check which they don't explain why they are doing it its the very specific about percents which I don't understand.

```java
 private static final Pattern PERCENTS_PAT = Pattern.compile("(%)([0-9a-fA-F])([0-9a-fA-F])");
```

should I add this check?

Reply:
OK I added the PERCENT check as well.